src/coreclr/jit/lowerxarch.cpp

```diff
@@ -2712,9 +2712,8 @@ void Lowering::LowerHWIntrinsicGetElement(GenTreeHWIntrinsic* node)
 
     node->SetSimdSize(16);
 
-    if (!varTypeIsFloating(simdBaseType))
+    if (resIntrinsic != intrinsicId)
     {
-        assert(node->GetHWIntrinsicId() != intrinsicId);
```

Comment:
The intent of this assert was to catch infinite recursion on `node`. Probably be good to keep it.

Reply:
I deleted it because `node->GetHWIntrinsicId() == resIntrinsic`. If there is concern about `resIntrinsic` starting to mean something different, I think we can reformulate the `if` as `node->GetHWIntrinsicId() != intrinsicId` instead.

Reply:
I think I'd prefer that check. It won't really be any more expensive in practice but its overall safer against refactorings/changes